packages/medusa/src/services/totals.ts

```diff
@@ -309,9 +305,31 @@ class TotalsService extends TransactionBaseService {
    */
   getShippingTotal(cartOrOrder: Cart | Order): number {
     const { shipping_methods } = cartOrOrder
-    return shipping_methods.reduce((acc, next) => {
-      return acc + next.price
-    }, 0)
+    return (
+      shipping_methods?.reduce((acc, shippingMethod) => {
+        if (
+          this.featureFlagRouter_.isFeatureEnabled(
+            TaxInclusivePricingFeatureFlag.key
+          ) &&
+          shippingMethod.includes_tax
+        ) {
+          const rate =
```

Comment:
Not sure if I understood it. In case we don't have tax_lines, the taxes will be 0, and free shipping will have price 0 no?

Reply:
if we have a free shipping discount then it will be 0, but I am wondering why it is not taken into account for the shipping total. specially that the discount total, filter on the discounts that are not free shipping.

also, if there is not tax lines it could be due to the back compatibility of the tax rate being setup on the cartOrOrder, but it is not taken into account. now that there is the inclusive tag, the best way to know the real price of the shipping method would be to call the method `getShippingMethodTotals` that take care of the different circumstances (plus the comment on the method too). that way you can compute back the real price. it is not the more performant at the moment, but until it is refactored i think it is the best approach, a bit like i ve done it here `getLineItemTotal`

@srindom could you confirm if i am missing something just for that part please?

Reply:
Just to weigh in here:

Think it is a great idea to change this method to just be:
   ```js
    const { shipping_methods } = cartOrOrder
    return shipping_methods.reduce((acc, next) => {
      const totals = this.getShippingMethodTotals(next, cartOrOrder)
      return acc + totals.subtotal
    }, 0)
   ```
   
It will then be `getShippingMethodTotals` that is responsible for doing the tax calculation stuff.

Not sure if that was what you meant @adrien2p?

Reply:
Yes this is exactly what i meant 😊 thanks man

Reply:
But surrounded by a lot of explaination on why that choice which might be less understandable in the end 🤔

Reply:
Haha alright :D 

@carlos-r-l-rodrigues - does this make sense to you?